src/Web/Rsp.IrasPortal.Web/Controllers/QuestionnaireController.cs

```diff
@@ -349,11 +351,20 @@ public async Task<IActionResult> SaveResponses(QuestionnaireViewModel model, str
             // otherwise resume from the NextStage in sequence
             return RedirectToAction(nameof(Resume), new
             {
-                applicationId = application.ApplicationId, categoryId = navigation.NextCategory, sectionId = navigation.NextStage
-
+                applicationId = application.ApplicationId,
+                categoryId = navigation.NextCategory,
+                sectionId = navigation.NextStage
             });
         }
 
+        if (saveForLater == bool.TrueString)
+        {
+            TempData["ProjectTitle"] = HttpContext.Session.GetString("ShortProjectTitle");
+            TempData["CategoryId"] = navigation.NextCategory;
+            TempData["ApplicationId"] = application.ApplicationId;
```

Comment:
As mentioned earlier, please define these keys in `TempDataKeys` class as constants and follow the naming convention. This way you only define it once and reuse wherever needed, instead of hardcoding.
```suggestion
            TempData[TempDataKeys.ShortProjectTitle] = shortProjectTitle; // please see earlier comment on line 252
            TempData[TempDataKeys.CategoryId] = navigation.CurrentCategory;
            TempData[TempDataKeys.ApplicationId] = application.ApplicationId; 
```